geode-core/src/main/java/org/apache/geode/internal/cache/GridAdvisor.java

```diff
@@ -432,6 +447,17 @@ public boolean equals(Object obj) {
         if (this.gp.getPort() == other.gp.getPort()) {
           final String thisHost = this.gp.getHost();
           final String otherHost = other.gp.getHost();
+          final String thisInternalHost = this.gp.getInternalHost();
```

Comment:
If internalHost is being included in `equals()` calculations, it should probably also be included in hashCode()`.